akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocketClientBlueprint.scala

```diff
@@ -54,7 +56,8 @@ private[http] object WebSocketClientBlueprint {
 
     val valve = StreamUtils.OneTimeValve()
 
-    val (initialRequest, key) = Handshake.Client.buildRequest(uri, extraHeaders, subprotocol.toList, settings.websocketRandomFactory())
+    val subprotocols: immutable.Seq[String] = subprotocol.toList.flatMap(_.split(",")).map(_.trim)
```

Comment:
From the RFC, "If present, this value indicates one
        or more comma-separated subprotocol the client wishes to speak,
        ordered by preference.  The elements that comprise this value
        MUST be non-empty strings with characters in the range U+0021 to
        U+007E not including separator characters as defined in
        [RFC2616] and MUST all be unique strings" - that suggests comma's are allowed inside the protocol names (U+002C), but how does that work when the values are also comma-separated?

Reply:
it says `not including separator characters as defined in
[RFC2616]`, and I think a comma is a part of this definition:

> separators     = "(" | ")" | "<" | ">" | "@"
                      | "," | ";" | ":" | "\" | <">
                      | "/" | "[" | "]" | "?" | "="
                      | "{" | "}" | SP | HT

so, seems like subprotocol cannot include commas

Reply:
Oops :D - that's clear then! :+1:

Reply:
Should we ignore all separator characters or just trim the whitespace?

Reply:
not sure. If we want to enforce the RFC, we can validate the `subprotocols` field and throw an exception if it contradicts the specification.
Although, this might break client code. Current  implementation is the closest to current behavior (not enforcing any rules).

Reply:
yeah, also the specification itself uses spaces in the examples, so it'd be silly to disallow that. I agree this seems fine.